pkg/plugin/plugin.go

```diff
@@ -22,9 +23,9 @@ type plugin struct {
 	clusterUpgrader         cluster.Upgrader
 	armGenerator            arm.Generator
 	configGenerator         config.Generator
-	apiValidator            *api.Validator
-	adminValidator          *api.Validator
-	pluginTemplateValidator *api.Validator
+	apiValidator            *validate.APIValidator
```

Comment:
I don't believe there is any value to having these fields in the plugin.  The validators are not interfaces nor do I think they immediately need to become so.  Refactor the validate.NewFooValidator() calls to the point where each validator is actually used.

Reply:
added as part of Validate main interfaces.